components/Pagination.tsx

```diff
@@ -0,0 +1,115 @@
+    baseUrl: string;
+    page: number;
+    search: string;
+    totalCount: number;
+    paginationLimit?: number;
+    postPerPage?: number;
+}
+
+export const Pagination = ({
+    baseUrl,
+    page,
+    search,
+    totalCount,
+    paginationLimit = 5,
+    postPerPage = 10
+}: PaginationProps): JSX.Element => {
+    const buttonClasses = [
+        'border',
+        'border-gray',
+        'flex',
+        'justify-center',
+        'no-underline',
+        'px-2',
+        'py-2',
+        'rounded',
+        'focus:outline-none'
+    ];
+    const buttonClassesActive = ['border-purple-200', 'text-purple-200'];
+    const arrowButtons = ['w-full', 'h-full', 'flex', 'items-center'];
+    const leftArrowClasses = ['transform', 'rotate-180'];
+    const url = `${baseUrl}/${search}`;
+    // Pagination
+    const totalPages = Math.ceil(totalCount / postPerPage);
+    const pagStart = PaginationStart(page, totalPages, paginationLimit);
+    const pagEnd = totalPages > paginationLimit ? paginationLimit : totalPages;
+    const buttonCount = new Array(pagEnd).fill(0);
+    return (
+        <>
+            <ul className="list-none flex justify-center">
+                {page > 1 && (
+                    <li className="mr-4 w-8">
+                        <Link href={`${url}/${page - 1}`}>
+                            <a className={classNames(buttonClasses, arrowButtons)}>
+                                <ArrowSVG className={leftArrowClasses} />
+                            </a>
+                        </Link>
+                    </li>
+                )}
+                {buttonCount.map((item, index) => {
+                    const pagNumber = pagStart + index;
+                    return (
+                        <li key={index} className="mr-4 w-8">
+                            <Link href={`${url}/${pagNumber}`}>
+                                <a
+                                    className={classNames(
+                                        buttonClasses,
+                                        page == pagNumber && buttonClassesActive
+                                    )}
+                                >
+                                    {pagNumber}
+                                </a>
+                            </Link>
+                        </li>
+                    );
+                })}
+                {page < totalPages && (
+                    <li className="mr-4 w-8">
+                        <Link href={`${url}/${page + 1}`}>
+                            <a className={classNames(buttonClasses, arrowButtons)}>
+                                <ArrowSVG />
+                            </a>
+                        </Link>
+                    </li>
+                )}
+            </ul>
+        </>
+    );
+};
+
+const PaginationStart = (page: number, totalPages: number, limit: number): number => {
+    const halfLimit = Math.floor(limit / 2);
+    if (totalPages < limit || page <= halfLimit) {
```

Comment:
Is good practice to put this conditions in a function with a name that explain what it does.